src/Regulator.cpp

```diff
@@ -134,14 +282,25 @@ Regulator::Regulator(int rcvChannels, int bit_res, int FPP, int qLen, int bqLen,
     , m_b_BroadcastRingBuffer(NULL)
     , m_b_BroadcastQueueLength(bqLen)
 {
-    // catch settings that are compute bound using long HIST
-    // hub client rcvChannels is set from client's settings parameters
-    // hub server rcvChannels is set from connecting client, not from hub parameters
-    //    if (mNumChannels > MaxChans) {
-    //        std::cerr << "*** Regulator.cpp: receive channels = " << mNumChannels
-    //                  << " larger than max channels = " << MaxChans << "\n";
-    //        exit(1);
-    //    }
+    if (gVerboseFlag)
+        cout << "mPacketsInThePast = " << mPacketsInThePast << " at " << mFPP << "\n";
+
+    mPcnt = 0;
+    mTime = new Time();
+    mTime->start();
+    mUpToNow   = mPacketsInThePast * mFPP;        // duration
```

Comment:
seems like all of these should be initialized in `setFPPratio` and use `mPeerFPP` instead?